### PR TITLE
rustfmt doc code blocks with multiple comma-separated attributes

### LINKED ISSUE #3158: document code block can be annotated with several attributes
The `CodeBlockAttribute` enum needs to be updated to support attributes separated with commas, e.g.,:

```rust
// Rust and ignore attributes:
// ```rust,ignore
// foo! { x, y }
// ```
```

### PR DESCRIPTION
**Pull request by nipunn1313**
Added test covering this. Chose to treat the code block
as rust if and only if all of the comma-separated attributes
are rust-valid. Chose to allow/preserve whitespace around commas

Fixes #3158

### CODE REVIEW

**Comment by calebcartwright** [comment]
I left a note over in #4733 but there's also the wrinkle of the 1.x vs. 2.x that will definitely be a factor

**Comment by nipunn1313** [comment]
I merged in @davidBar-On 's ideas of
1) adding more unit test cases around empty attribute substrings
2) Having `compile_fail` code not format.

File: src/comment.rs [hunk 1]
```diff
@@ -392,28 +392,25 @@ fn identify_comment(
...
-            "no_run" => CodeBlockAttribute::NoRun,
-            "compile_fail" => CodeBlockAttribute::CompileFail,
-            _ => CodeBlockAttribute::Text,
+    /// Parse comma separated attributes list. Return rust only if all
+    /// attributes are valid rust attributes
+    /// See https://doc.rust-lang.org/rustdoc/print.html#attributes
+    fn new(attributes: &str) -> CodeBlockAttribute {
+        for attribute in attributes.split(",") {
+            match attribute.trim() {
+                "" | "rust" | "should_panic" | "no_run" | "edition2015" | "edition2018" => (),
```
Comment (nipunn1313) [inline src/comment.rs:408]: should we proactively add `edition2021`? Seems inevitable it will appear https://doc.rust-lang.org/rustdoc/print.html#attributes soon enough?

Comment (calebcartwright) [reply src/comment.rs:408]: yeah let's do. we've already got 2021 included in other edition related areas so might as well get this updated too

File: src/comment.rs [hunk 2]
```diff
@@ -647,23 +644,28 @@ impl<'a> CommentRewrite<'a> {
...
+                                } else {
+                                    trim_custom_comment_prefix(&self.code_block_buffer)
+                                }
                             } else {
                                 trim_custom_comment_prefix(&self.code_block_buffer)
                             }
-                        } else {
-                            trim_custom_comment_prefix(&self.code_block_buffer)
                         }
                     }
```
Comment (calebcartwright) [inline src/comment.rs:670]: Overall changes look good to me. Somewhat of a nit, but I think if we were to move some of the conditionals up as guards on arms then we could reduce the nested conditional complexity and accompanying indentation levels, example below. What do you think?

```rust
CodeBlockAttribute::Rust if self.code_block_buffer.is_empty() => String::new(),
CodeBlockAttribute::Rust if config.format_code_in_doc_comments() => {
    let mut config = self.fmt.config.clone();
    // ....
}
CodeBlockAttribute::Rust => trim_custom_comment_prefix(&self.code_block_buffer),

**Review by calebcartwright — APPROVED** [review]
Awesome thank you!

File: src/comment.rs [hunk 3]
```diff
@@ -647,25 +645,21 @@ impl<'a> CommentRewrite<'a> {
...
-                            }
+                        if let Some(s) =
+                            crate::format_code_block(&self.code_block_buffer, &config, false)
+                        {
+                            trim_custom_comment_prefix(&s.snippet)
                         } else {
                             trim_custom_comment_prefix(&self.code_block_buffer)
                         }
                     }
+                    _ => trim_custom_comment_prefix(&self.code_block_buffer),
```
Comment (nipunn1313) [inline src/comment.rs:662]: What do you think of this? Limited to just two arms - since I found that many of the arms from before fell to the same case.

It's basically an if-else at this point (if we should be formatting, format, else trim_custom_comment_prefix)

Comment (calebcartwright) [reply src/comment.rs:662]: guess the only difference really becomes the call to trim the prefix on an empty string which should be a functional no op.

should be fine
